File: demo/cpp/OneTimeSetUpTests.cpp
#include "ExpectedOutcomeAttributes.h"

using namespace System::Collections::Generic;
using namespace System::Text;
using namespace NUnit::Framework;

namespace NUnitTestDemo
{
    [TestFixture, ExpectPass]
    public ref class OneTimeSetUpTests
    {
	private:
        int SetUpCount;
        int TearDownCount;

	public:
        [OneTimeSetUp]
        void BeforeTests()
        {
            Assert::That(SetUpCount, Is::EqualTo(0));
            Assert::That(TearDownCount, Is::EqualTo(0));
            SetUpCount++;
		}

        [OneTimeTearDown]
        void AfterTests()
        {
            Assert::That(SetUpCount, Is::EqualTo(1), "Unexpected error");
            Assert::That(TearDownCount, Is::EqualTo(0));
            TearDownCount++;
        }

        [Test]
        void Test1()
        {
            Assert::That(SetUpCount, Is::EqualTo(1));
            Assert::That(TearDownCount, Is::EqualTo(0));
        }

        [Test]
        void Test2()
        {
            Assert::That(SetUpCount, Is::EqualTo(1));
            Assert::That(TearDownCount, Is::EqualTo(0));
        }
	};
}

File: demo/vb/OneTimeSetUpTests.vb
Imports NUnit.Framework

Namespace NUnitTestDemo

    <TestFixture, ExpectPass>
    Public Class OneTimeSetUpTests

        Dim SetUpCount As Integer
        Dim TearDownCount As Integer

        <OneTimeSetUp>
        Public Sub BeforeTests()
            Assert.That(SetUpCount, Iz.EqualTo(0))
            Assert.That(TearDownCount, Iz.EqualTo(0))
            SetUpCount = SetUpCount + 1
        End Sub

        <OneTimeTearDown>
        Public Sub AfterTests()
            Assert.That(SetUpCount, Iz.EqualTo(1), "Unexpected error")
            Assert.That(TearDownCount, Iz.EqualTo(0))
            TearDownCount = TearDownCount + 1
        End Sub

        <Test>
        Public Sub Test1()
            Assert.That(SetUpCount, Iz.EqualTo(1))
            Assert.That(TearDownCount, Iz.EqualTo(0))
        End Sub

        <Test>
        Public Sub Test2()
            Assert.That(SetUpCount, Iz.EqualTo(1))
            Assert.That(TearDownCount, Iz.EqualTo(0))
        End Sub

    End Class

End Namespace